conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureDoubleList.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib.internal;
+
+import com.palantir.logsafe.exceptions.SafeUnsupportedOperationException;
+import java.util.AbstractList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.RandomAccess;
+import org.eclipse.collections.impl.list.mutable.primitive.DoubleArrayList;
+
+public final class ConjureDoubleList extends AbstractList<Double> implements RandomAccess {
```

Comment:
This is weird. eclipse-collections doesn't extend AbstractList<> so I need to use this delegating pattern. I guess this takes 8 more bytes... FastUtils does extend this (https://github.com/karussell/fastutil/blob/master/src/it/unimi/dsi/fastutil/doubles/DoubleArrayList.java#L69), but its no longer maintained... Ideally the implementation already extends this class so we can instantiate directly in conjureCollections. This feels really close to writing our own impl.

Curious on your thoughts about if its worth pivoting to us writing the implementation where we throw on all unused code paths (everything except add, addAll) will prevent unintentionally picking up a inefficient default impl.